libvips/foreign/vips2webp.c

```diff
@@ -82,6 +83,73 @@ get_preset( VipsForeignWebpPreset preset )
 	return( -1 );
 }
 
+#if WEBP_ENCODER_ABI_VERSION < 0x0100
+typedef struct {
+	uint8_t *mem;
+	size_t size;
+	size_t max_size;
+	uint32_t pad[1];
+} WebPMemoryWriter;
+
+static void WebPMemoryWriterInit( WebPMemoryWriter *writer ) {
+	writer->mem = NULL;
+	writer->size = 0;
+	writer->max_size = 0;
+}
+
+static int WebPMemoryWrite( const uint8_t *data, size_t data_size,
+	const WebPPicture *picture ) {
+	WebPMemoryWriter* const w = (WebPMemoryWriter*) picture->custom_ptr;
+	size_t next_size;
+
+	if (w == NULL) {
+		return 1;
+	}
+
+	next_size = w->size + data_size;
+
+	if( next_size > w->max_size ) {
+		uint8_t *new_mem;
+		size_t next_max_size = w->max_size * 2;
+
+		if( next_max_size < next_size )
+			next_max_size = next_size;
+		if( next_max_size < 8192 )
+			next_max_size = 8192;
+
+		new_mem = (uint8_t*) malloc( next_max_size );
```

Comment:
Do you mind replacing the default memory writer from libwebp with a vips specific one? Then we could use that for all supported libwebp versions. You should look at 0.5.0 for reference, it has a newer version that uses uint64_t internally instead of size_t.